docs/guides/migrate-from-buidler.md

```diff
@@ -0,0 +1,180 @@
+}
+```
+
+you need to replace it with:
+
+```js
+networks: {
+  hardhat: {
+    blockGasLimit: 12000000
+  }
+}
+```
+
+### Updating your Solidity config
+
+Hardhat has native support for multiple versions of Solidity. This requires a few changes in how you customize your
+`solc` setup.
+
+If you had something like this in your config:
+
+```js
+solc: {
+  version: "0.7.1",
+  optimizer: {
+    enabled: true
+  }
+}
+```
+
+You should change it to this:
+
+```js
+solidity: {
+  version: "0.7.1",
+  settings: {
+    optimizer: {
+      enabled: true
+    }
+  }
+}
+```
+
+This is a very simple config, but Hardhat supports arbitrarily complex compilation setups,
+down to the individual file level. Take a look at [this guide](./compile-contracts.md) to learn more about it.
+
+## TypeScript support changes
+
+Hardhat enables its TypeScript support when your config is written in TypeScript and ends in `.ts`. If you were using
+TypeScript, but a JavaScript config, please take a look at [this guide](./typescript.md).
+
+You don't need a `tsconfig.json` file with Hardhat. We recommend you delete it.
+
+If you prefer to keep it, you should base it on the template presented [here](./typescript.md#). Just make
+sure you removed all the `type-extension.d.ts` files, and keep your config file in your `files` field.
+
+## Importing artifacts and ambiguous names
+
+Hardhat supports multiple contracts with the same name.
+
+If you have contracts which share their name, you can't import their artifacts using just the name. 
+
+For example, if you have a contract named `Ownable`, and one of your dependencies has a contract with the same name, you won't be able to do
+`artifacts.require("Ownable")` nor `ethers.getContractFactory("Ownable")`. You need to use the contract's Fully
+Qualified Name instead (e.g. `contracts/Ownable.sol:Ownable`).
+
+If you try to import a contract with a repeated name, Hardhat will fail, with an error that includes the different
+options to fix it. All you need to do is copy & paste them.
+
+For example, you may need to replace this
+
+```js
+const Ownable = await ethers.getContractFactory("Ownable");
+```
+
+with this
+
+```js
+const Ownable = await ethers.getContractFactory("contracts/Ownable.sol:Ownable");
+```
+
+## Mocha and VSCode setup changes
+
+If you are running your tests directly with Mocha, or though a VSCode Mocha plugin, please take a look at [this
+updated guide](./vscode-tests.md).
+
+## Community plugins
+
+All of the official Buidler plugins have already been migrated to Hardhat.
+
+Some community-built plugins, haven't been migrated yet. If you are using one of those, you have to temporarily disable them.
+
+### buidler-deploy
+
+This plugin is already being ported to Hardhat, and close to being released.
+
+Join our `#buidler-deploy` channel on [Discord](/discord) to get news and help about it.
+
+### buidler-typechain
+
+The TypeChain plugin hasn't been migrated yet, but you can run TypeChain directly like this:
```

Comment:
we need to remember to udpate this as soon as this is migrated

Reply:
Yep, this entire section.